cmake/ProcessOptions.cmake

```diff
@@ -304,7 +267,7 @@ function( NEST_PROCESS_WITH_READLINE )
   if ( with-readline )
     if ( NOT ${with-readline} STREQUAL "ON" )
       # a path is set
-      set( READLINE_ROOT_DIR "${with-readline}" )
```

Comment:
I don't see why are these changes actually necessary. Did something in CMake itself require these changes?

Reply:
During debugging work on LDTL loading, I found that our `-Dwith-ldtl` switch did not allow us to give a specific ltdl as argument, because we were setting the wrong variables: They need to be `<Packagename>_ROOT` (see https://cmake.org/cmake/help/latest/envvar/PackageName_ROOT.html) and newer CMAKEs (since 3.27) handle also uppercased package names (see https://cmake.org/cmake/help/latest/policy/CMP0144.html#policy:CMP0144)